clients/src/test/java/org/apache/kafka/clients/ClientUtilsTest.java

```diff
@@ -96,10 +100,32 @@ public void testParseAndValidateAddressesWithReverseLookup() {
         }
     }
 
-    @Test
-    public void testInvalidConfig() {
+    static Stream<List<String>> provideValidBrokerAddressTestCases() {
+        return Stream.of(
+            Arrays.asList("localhost:9997", "localhost:9998", "localhost:9999"),
+            // Intentionally provide a single string, as users may provide space-separated brokers, which will be parsed as a single string.
+            Arrays.asList("localhost:9997 localhost:9998 localhost:9999")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideValidBrokerAddressTestCases")
+    public void testValidBrokerAddress(List<String> addresses) {
+        assertDoesNotThrow(() -> ClientUtils.parseAndValidateAddresses(addresses, ClientDnsLookup.USE_ALL_DNS_IPS));
+    }
+
+    static Stream<List<String>> provideInvalidBrokerAddressTestCases() {
+        return Stream.of(
+            Collections.singletonList("localhost:9997\nlocalhost:9998\nlocalhost:9999"),
+            Collections.singletonList("localhost:10000")
```

Comment:
`localhost:10000` is valid, and the test can cause "expected" error due to "random.value"